scapy/layers/inet.py

```diff
@@ -2053,6 +2053,150 @@ def stop_ack_timeout(self):
         raise self.CLOSED()
 
 
+class TCP_server(TCP_client):
+    """
+    Creates a TCP Server Automaton.
+    This automaton will handle TCP 3-way handshake.
+    """
+    def parse_args(self, ip, sport, *args, **kargs):
+        from scapy.sessions import TCPSession
+        self.dst = str(Net(ip))
+        self.sport = sport
+        self.l4 = IP(src=ip) / TCP(sport=self.sport, flags=0,
+                                   seq=random.randrange(0, 2**32))
+        self.src = self.l4.src
+        self.sack = self.l4[TCP].ack
```

Comment:
This is a bit cheap: you handled only a single client at the time. I must say I envisioned something a bit more useful, handling multiple streams.
Did you have a look at Transmission Control Block?